WalletWasabi.Fluent/ViewModels/Wallets/Home/History/HistoryViewModel.cs

```diff
@@ -290,22 +294,85 @@ private IEnumerable<HistoryItemViewModelBase> GenerateHistoryList(List<Transacti
 			}
 
 			if (coinJoinGroup is { } cjg &&
-				((i + 1 < txRecordList.Count && !txRecordList[i + 1].IsOwnCoinjoin) || // The next item is not CJ so add the group.
-				 i == txRecordList.Count - 1)) // There is no following item in the list so add the group.
+				((i + 1 < summaries.Count && !summaries[i + 1].IsOwnCoinjoin) || // The next item is not CJ so add the group.
+				 i == summaries.Count - 1)) // There is no following item in the list so add the group.
 			{
 				if (cjg.CoinJoinTransactions.Count == 1)
 				{
 					var singleCjItem = new CoinJoinHistoryItemViewModel(UiContext, cjg.OrderIndex, cjg.CoinJoinTransactions.First(), _walletVm, balance, true);
-					yield return singleCjItem;
+					history.Add(singleCjItem);
 				}
 				else
 				{
 					cjg.SetBalance(balance);
-					yield return cjg;
+					history.Add(cjg);
 				}
 
 				coinJoinGroup = null;
 			}
 		}
+
+		// This second iteration is necessary to transform the flat list of speed-ups into actual groups.
+		// Here are the steps:
+		// 1. Identify which transactions are CPFP (parents) and their children.
+		// 2. Create a speed-up group with parent and children.
+		// 3. Remove the previously added items from the history (they should no longer be there, but in the group).
+		// 4. Add the group.
+		foreach (var summary in summaries)
+		{
+			if (summary.Transaction.IsCPFPd)
+			{
+				// Group creation.
+				var childrenTxs = summary.Transaction.ChildrenPayForThisTx;
+
+				if (!TryFindHistoryItem(summary.TransactionId, history, out var parent))
+				{
+					continue; // If the parent transaction is not found, continue with the next summary.
+				}
+
+				var groupItems = new List<HistoryItemViewModelBase> { parent };
+				foreach (var childTx in childrenTxs)
+				{
+					if (TryFindHistoryItem(childTx.GetHash(), history, out var child))
+					{
+						groupItems.Add(child);
+					}
+				}
+
+				// If there is only one item in the group, it's not a group.
+				// This can happen, for example, when CPFP occurs between user-owned wallets.
+				if (groupItems.Count <= 1)
+				{
+					continue;
+				}
+
+				groupItems.ForEach(x => x.IsChild = true);
```

Comment:
I think setting this flag should happen in the group object's constructor.
I don't insist on it, just consider.